tests/test_read_write/test_write_csv.py

```diff
@@ -0,0 +1,118 @@
+import pytest
+from clumper import Clumper
+
+
+def test_write_csv(tmp_path):
+    d = tmp_path / "sub"
+    d.mkdir()
+    path = d / "monopoly_copy.csv"
+    Clumper.read_csv("tests/data/monopoly.csv").write_csv(path)
+    reader = Clumper.read_csv(path)
+    assert Clumper.read_csv("tests/data/monopoly.csv").collect() == reader.collect()
+
+
+def test_write_csv_append_mode(tmp_path):
+    """Test that the append mode works."""
+    d = tmp_path / "sub"
+    d.mkdir()
+    path = d / "monopoly_copy.csv"
+    Clumper.read_csv("tests/data/monopoly.csv").write_csv(path)
+    Clumper.read_csv("tests/data/monopoly.csv").write_csv(path, mode="a")
+    reader = Clumper.read_csv(path)
+    # the header is appended as well, hence the +1
+    assert len(reader) == (2 * len(Clumper.read_csv("tests/data/monopoly.csv"))) + 1
+
+
+@pytest.fixture()
+def data():
+    return [
+        {"One": "a", "Two": "1", "Three": "one"},
+        {"One": "b", "Two": "2", "Three": "two"},
+        {"One": "", "Two": "3", "Three": "three"},
```

Comment:
Could we add an example like: 

```
{"Two": 3, "Three": "three"}
```

That feels like an interesting edge case to make sure that we get right.

Reply:
The only way with this is to that all values assigned to two are either strings or numbers, not a mix. If it is a mix, then it comes back as a string type.